desktop_version/src/DiscordNetwork.c

```diff
@@ -0,0 +1,188 @@
+#include "MakeAndPlay.h"
+
+#ifndef MAKEANDPLAY
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <SDL.h>
+
+#include "Vlogging.h"
+
+// Library includes
+#include "discord_game_sdk.h"
+
+#if defined(_WIN32)
+#define DISCORD_LIBRARY "discord_game_sdk.dll"
+#elif defined(__APPLE__)
+#define DISCORD_LIBRARY "libdiscord_game_sdk.dylib"
+#elif defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
+#define DISCORD_LIBRARY "libdiscord_game_sdk.so"
+#else
+#error DISCORD_LIBRARY: Unrecognized platform!
+#endif
+
+
+#define DISCORD_CLIENT_ID 1315544357532729447
+
+// TO TERRY/FLIBIT: You can create your own Discord instance at the Discord Developer Portal. This ID belongs to me, so just be aware that if my account was to get hacked, VVVVVV RPC would too. Use your own!
+
+
+struct DISCORD_application {
+    struct IDiscordCore* core;
+    struct IDiscordActivityManager* activityMan;
+} app;
+
+static struct DiscordActivity activity;
+
+static bool discordDetected = false;
+static bool discordPostInit = false;
+
+
+static void* libHandle = NULL;
+
+
+
+#define FUNC_LIST \
+    FOREACH_FUNC(enum EDiscordResult, DiscordCreate, (DiscordVersion, struct DiscordCreateParams*, struct IDiscordCore**))
+
+
+
+#define FOREACH_FUNC(rettype, name, params) static rettype (*name) params = NULL;
+FUNC_LIST
+#undef FOREACH_FUNC
+
+bool DISCORD_REQUIRE(int x)
+{
+    if (!discordDetected && discordPostInit)
+    {
+        return false;
+    }
+    if (x != DiscordResult_Ok)
+    {
+        return false;
+    }
+    return true;
+}
+
+
+
+static void ClearPointers(void)
+{
+    SDL_UnloadObject(libHandle);
+    libHandle = NULL;
+#define FOREACH_FUNC(rettype, name, params) name = NULL;
+    FUNC_LIST
+#undef FOREACH_FUNC
+}
+
+void DISCORD_shutdown(void)
+{
+    if (libHandle != NULL)
+    {
+        app.core->destroy(app.core);
+        ClearPointers();
+    }
+}
+
+int32_t DISCORD_init(void)
+{
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
```

Comment:
This should probably be rewritten to check for OSes that Discord *does* support, rather than ones it doesn't.